src/Command/CodeStudio/CodeStudioWizardCommand.php

```diff
@@ -0,0 +1,622 @@
+    foreach ($required_permissions as $name) {
+      if (!array_key_exists($name, $keyed_permissions)) {
+        throw new AcquiaCliException("The Acquia Cloud account {account} does not have the required '{name}' permission. Please add the permissions to this user or use an API Token belonging to a different Acquia Cloud user.", [
+          'account' => $account->mail,
+          'name' => $name
+        ]);
+      }
+    }
+  }
+
+  /**
+   * @param array $project
+   * @param string $project_access_token_name
+   *
+   * @return mixed
+   */
+  protected function createProjectAccessToken(array $project, string $project_access_token_name) {
+    $this->io->writeln("Creating project access token...");
+
+    if ($existing_token = $this->getGitLabProjectAccessTokenByName($project, $project_access_token_name)) {
+      $this->checklist->addItem("Deleting access token named <comment>$project_access_token_name</comment>");
+      $this->gitLabClient->projects()
+            ->deleteProjectAccessToken($project['id'], $existing_token['id']);
+      $this->checklist->completePreviousItem();
+    }
+    $this->checklist->addItem("Creating access token named <comment>$project_access_token_name</comment>");
+    $project_access_token = $this->gitLabClient->projects()
+          ->createProjectAccessToken($project['id'], [
+          'name' => $project_access_token_name,
+          'scopes' => ['api', 'write_repository'],
+        ]);
+    $this->checklist->completePreviousItem();
+    return $project_access_token['token'];
+  }
+
+  /**
+   * @param array $project
+   * @param string $cloud_application_uuid
+   * @param string $cloud_key
+   * @param string $cloud_secret
+   * @param string $project_access_token_name
+   * @param string $project_access_token
+   */
+  protected function setGitLabCiCdVariables(array $project, string $cloud_application_uuid, string $cloud_key, string $cloud_secret, string $project_access_token_name, string $project_access_token): void {
+    $this->io->writeln("Setting GitLab CI/CD variables for {$project['path_with_namespace']}..");
+    $gitlab_cicd_variables = [
+      [
+        'key' => 'ACQUIA_APPLICATION_UUID',
+        'value' => $cloud_application_uuid,
+        'masked' => FALSE,
+        'protected' => FALSE
+      ],
+      [
+        'key' => 'ACQUIA_CLOUD_API_TOKEN_KEY',
+        'value' => $cloud_key,
+        'masked' => FALSE,
+        'protected' => FALSE
+      ],
+      [
+        'key' => 'ACQUIA_CLOUD_API_TOKEN_SECRET',
+        'value' => $cloud_secret,
+        'masked' => FALSE,
+        'protected' => FALSE
+      ],
+      [
+        'key' => 'ACQUIA_GLAB_TOKEN_NAME',
+        'value' => $project_access_token_name,
+        'masked' => FALSE,
+        'protected' => FALSE
+      ],
+      [
+        'key' => 'ACQUIA_GLAB_TOKEN_SECRET',
+        'value' => $project_access_token,
+        'masked' => TRUE,
+        'protected' => FALSE
+      ],
+      [
+        'key' => 'ACQUIA_CLOUD_SSH_KEY',
```

Comment:
This needs to be stored in the vault as a secret. @msherron do you know how the user will add this as a secret to the vault?

Reply:
Separate related question, why don't we use the "file" variable type in GitLab?